File: apps/api/src/common/health.controller.ts
import { Controller, Get } from '@nestjs/common';
import { InjectDataSource } from '@nestjs/typeorm';
import { DataSource } from 'typeorm';

@Controller('health')
export class HealthController {
  constructor(@InjectDataSource() private readonly dataSource: DataSource) {}

  @Get()
  async check() {
    // Kiểm tra cả PostGIS, không chỉ kết nối: thiếu extension thì mọi truy vấn
    // geo sẽ fail lúc chạy chứ không phải lúc khởi động.
    const [{ postgis_version }] = await this.dataSource.query<
      { postgis_version: string }[]
    >('SELECT postgis_version() AS postgis_version');

    return { status: 'ok', database: 'up', postgis: postgis_version };
  }
}

File: src/Massage.Api/Modules/Health/HealthController.cs
using Massage.Api.Data;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace Massage.Api.Modules.Health;

/// <summary>Kiểm tra tình trạng hệ thống.</summary>
[ApiController]
[Route("health")]
[Tags("Health")]
public class HealthController(AppDbContext db) : ControllerBase
{
    /// <summary>Trạng thái API, kết nối DB và phiên bản PostGIS.</summary>
    [HttpGet]
    public async Task<IActionResult> Get(CancellationToken ct)
    {
        // Kiểm tra cả PostGIS chứ không chỉ kết nối: thiếu extension thì mọi truy vấn
        // geo sẽ hỏng lúc chạy chứ không phải lúc khởi động, và health vẫn báo xanh.
        var postgis = await db.Database
            .SqlQuery<string>($"SELECT postgis_version() AS \"Value\"")
            .FirstAsync(ct);

        return Ok(new { status = "ok", database = "up", postgis });
    }
}
